xbmc/filesystem/CurlFile.cpp

```diff
@@ -1097,7 +1097,7 @@ bool CCurlFile::Open(const CURL& url)
 
   m_httpresponse = m_state->Connect(m_bufferSize);
 
-  if (m_httpresponse <= 0 || (m_failOnError && m_httpresponse >= 400))
+  if (m_httpresponse < 0 || (m_failOnError && m_httpresponse >= 400))
```

Comment:
My concern with this is the history. https://github.com/xbmc/xbmc/pull/7290
it was specifically added to test specifically 0 error code due to things potentially not returning.

I dont know why this is required for this PR (i havent looked into it), but i think another solution may be necessary to take the original intent/use into consideration

Reply:
curl returns HTTP response code 0 for file protocol. Maybe worth adding check allowing 0 for file:// specifically ?

Reply:
yeah, thats one way i had thought might be viable. havent looked any further into surrounding code though

Reply:
OK, added the guardrail for file protocol!